classes/QueryCondition.cls

```diff
@@ -0,0 +1,143 @@
+    public enum Operator {
+        EQUALS,
+        NOT_EQUALS,
+        LESS_THAN,
+        LESS_THAN_OR_EQUAL,
+        GREATER_THAN,
+        GREATER_THAN_OR_EQUAL
+    }
+
+    private final String fieldName;
+    private final Operator operator;
+    private final List<Object> values;
+
+    private static Boolean isSet = false;
+
+    public QueryCondition(String fieldName, Operator operator, List<Object> values) {
+        this.fieldName = fieldName;
+        this.operator = operator;
+        this.values = values;
+    }
+
+    public QueryCondition(String fieldName, Operator operator, Object value) {
+        this(fieldName, operator, new List<Object>{ value });
+    }
+
+    public QueryCondition(SObjectField field, Operator operator, Object value) {
+        this(field, operator, new List<Object>{ value });
+    }
+
+    public QueryCondition(SObjectField field, Operator operator, List<Object> values) {
+        this(field.getDescribe().getName(), operator, values);
+    }
+
+    public override String toString() {
+        String valueString = this.convertToString(this.values);
+        return this.fieldName + ' ' + this.getOperator() + ' ' + valueString;
+    }
+
+    private String convertToString(Object value) {
+        if (value == null) {
+            return 'null';
+        } else if (value instanceof Datetime) {
+            return this.formatDatetime(value);
+        } else if (value instanceof List<Object>) {
+            return this.formatList(value);
+        } else if (value instanceof String) {
+            String input = (String) value;
+            return '\'' + String.escapeSingleQuotes(input) + '\'';
+        }
+
+        String valueString = String.valueOf(value);
+
+        //checking for set since instance of Set is not supported
+        if (valueString.startsWith('{') && valueString.endsWith('}')) {
+            return this.formatSet(valueString);
+        }
+
+        return valueString;
+    }
+
+    private String formatDatetime(Object value) {
+        Datetime dt = (Datetime) value;
+        return dt.format(
+            'yyyy-MM-dd\'T\'HH:mm:ss\'Z\'',
+            'Greenwich Mean Time'
+        );
+    }
+
+    private String formatList(Object value) {
+        List<Object> values = (List<Object>) value;
+        List<String> innerStrings = new List<String>();
+
+        for (Object innerPred : values) {
+            // recurse for string value
+            String innerString = this.convertToString(innerPred);
+            innerStrings.add(innerString);
+        }
+
+        String start = innerStrings.size() > 1 ? '(' : '';
+        String ending = innerStrings.size() > 1 ? ')' : '';
+        
+        return start + String.join(innerStrings, ',') + ending;
+    }
+
+    private String formatSet(String valueString) {
+        List<String> setInner = valueString.substring(
+            1,
+            valueString.length() - 1
+        )
+        .split(',');
```

Comment:
you're going to want to split on `, ` instead. Sets add a space between values

Reply:
I opted with trimming the result when I ran into this, but that works too.